src/vs/workbench/contrib/performance/browser/performance.web.contribution.ts

```diff
@@ -7,23 +7,39 @@ import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle
 import { Registry } from 'vs/platform/registry/common/platform';
 import { Extensions, IWorkbenchContributionsRegistry } from 'vs/workbench/common/contributions';
 import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { posix } from 'vs/base/common/path';
+import { hash } from 'vs/base/common/hash';
 
 class ResourcePerformanceMarks {
 
 	constructor(@ITelemetryService telemetryService: ITelemetryService) {
 
-		type Entry = { name: string; duration: number };
+		type Entry = {
+			hosthash: string;
+			name: string;
+			duration: number;
+		};
 		type EntryClassifify = {
 			owner: 'jrieken';
 			comment: 'Resource performance numbers';
-			name: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'Resource name' };
+			hosthash: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'Hash of the hostname' };
+			name: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'Resource basename' };
 			duration: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; isMeasurement: true; comment: 'Resource duration' };
 		};
 		for (const item of performance.getEntriesByType('resource')) {
-			telemetryService.publicLog2<Entry, EntryClassifify>('startup.resource.perf', {
-				name: item.name,
-				duration: item.duration
-			});
+
+			try {
+				const url = new URL(item.name);
+				const name = posix.basename(url.pathname);
+
+				telemetryService.publicLog2<Entry, EntryClassifify>('startup.resource.perf', {
+					hosthash: `H${hash(url.host).toString(16)}`,
```

Comment:
We don't want to allowlist a certain set of hosts?

Reply:
yeah, we don't really want to know the exact host and I also don't want to hardcode a list of hostnames (in the OSS repo)